cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -217,7 +217,10 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
                         .build();
             }
 
-            status.setConditions(Collections.singletonList(readyCondition));
+            List<Condition> oldConditions = status.getConditions();
+            List<Condition> newConditions = oldConditions != null ? new ArrayList<>(oldConditions) : new ArrayList<>();
```

Comment:
Why do you create a new ArrayList and not just add it to the existing if it exists?

Reply:
The existing list is an unmodifiable list so we have to create a modifiable copy to be able to add the new status.

Reply:
Ok, makes sense.